library/src/main/java/com/bumptech/glide/load/engine/cache/InternalCacheDiskCacheFactory.java

```diff
@@ -2,47 +2,35 @@
 
 import android.content.Context;
 
-import com.bumptech.glide.Glide;
-
 import java.io.File;
 
 /**
  * Creates an {@link com.bumptech.glide.disklrucache.DiskLruCache} based disk cache in the internal disk cache
  * directory.
  */
-public final class InternalCacheDiskCacheFactory implements DiskCache.Factory {
-    private final Context context;
-    private final String diskCacheName;
-    private final int diskCacheSize;
+public final class InternalCacheDiskCacheFactory extends DiskLruCacheFactory {
 
-    public InternalCacheDiskCacheFactory(Context context, int diskCacheSize) {
-        this(context, null /*diskCacheName*/, diskCacheSize);
+    public InternalCacheDiskCacheFactory(Context context) {
+        this(context, DiskCache.Factory.DEFAULT_DISK_CACHE_DIR, DiskCache.Factory.DEFAULT_DISK_CACHE_SIZE);
     }
 
-    public InternalCacheDiskCacheFactory(Context context, String diskCacheName, int diskCacheSize) {
-        this.context = context;
-        this.diskCacheName = diskCacheName;
-        this.diskCacheSize = diskCacheSize;
+    public InternalCacheDiskCacheFactory(Context context, int diskCacheSize) {
+        this(context, DiskCache.Factory.DEFAULT_DISK_CACHE_DIR, diskCacheSize);
     }
 
-    @Override
-    public DiskCache build() {
-        DiskCache diskCache = null;
-        final File cacheDir;
-
-        if (diskCacheName != null) {
-            cacheDir = Glide.getPhotoCacheDir(context, diskCacheName);
-        } else {
-            cacheDir = Glide.getPhotoCacheDir(context);
-        }
-
-        if (cacheDir != null) {
-            diskCache = DiskLruCacheWrapper.get(cacheDir, diskCacheSize);
-        }
-
-        if (diskCache == null) {
```

Comment:
It's missing from the diff, but this has been added to the only place where build is called from in #374.